### PR TITLE
Post refactor - First PR

### PR DESCRIPTION
**Pull request by germa89**
Adds Solution with solution.py
Adds Selection, TimeFreqSelection, SpatialSelection with selection.py
Adds DataObject with data_object.py
Adds Mesh with mesh.py
Adds tools.select with tools.py

Adds temporary example and meeting notes in temp/

### CODE REVIEW

File: src/ansys/dpf/post/__init__.py [hunk 1]
```diff
@@ -21,14 +21,16 @@
...
-__version__ = importlib_metadata.version("ansys-dpf-post")
-
+from ansys.dpf.post import tools
 from ansys.dpf.post.common import Grouping as grouping
 from ansys.dpf.post.dpf_path import create_path_on_coordinates
 from ansys.dpf.post.misc import Report
 from ansys.dpf.post.post_utility import load_solution, print_available_keywords
 
+# this must be after some ansys.dpf.post import
+__version__ = importlib_metadata.version("ansys-dpf-post")
```
Comment (GuillemBarroso) [inline src/ansys/dpf/post/__init__.py:31]:
```suggestion
__version__ = importlib_metadata.version(__name__.replace(".", "-"))
```

File: src/ansys/dpf/post/post_utility.py [hunk 2]
```diff
@@ -17,24 +18,31 @@
...
+    legacy=True,
+):
     """Loads a solution and returns a :class:`ansys.dpf.post.Result` object.
 
     This class provides information on a given set on a given scoping.
 
     Parameters
     ----------
-    data_sources : str or ansys.dpf.core.DataSources
+    data_sources: str or ansys.dpf.core.DataSources
```
Comment (GuillemBarroso) [inline src/ansys/dpf/post/post_utility.py:33]:
```suggestion
    data_sources: str, ansys.dpf.core.DataSources
```

**Review by GuillemBarroso — APPROVED** [review]
LGTM. I just added some minor suggestions.

File: src/ansys/dpf/post/__init__.py (hunk 1 above)
Comment (PProfizi) [reply src/ansys/dpf/post/__init__.py:31]: @GuillemBarroso not sure why, but this did not work when importing in the docstring tests

Comment (GuillemBarroso) [reply src/ansys/dpf/post/__init__.py:31]: Yep... We had a similar problem when not hardcoding the name in pydpf-core as well. No idea why tho.

**Pull request merged**